packages/storybook/src/utilities/matrix.ts

```diff
@@ -41,7 +41,9 @@ export function cartesianProduct<T extends readonly unknown[]>(
         ...states: readonly unknown[]
     ): void => {
         if (currentDimensions && currentDimensions.length >= 1) {
-            const [currentDimension, ...remainingDimensions] = currentDimensions;
+            const [currentDimensionOrUndefined, ...remainingDimensions] = currentDimensions;
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+            const currentDimension = currentDimensionOrUndefined!;
```

Comment:
This one is super weird and I didn't figure it out. Typescript for some reason is now failing to infer that this value cannot be undefined when the storybook build runs (normally TS can see that we did a length check on the array so this is definitely not undefined). So I added the non-null assertion to help TS.

However the TS config as part of eslint-typescript (whish is the same tsconfig file) is still able to infer that the value is non-null so complains that we have an unnecessary non-null assertion. So I added an eslint disable for it.

Reply:
Can you add a brief comment to this effect (mostly to indicate that we thought about it, doesn't need to be exhaustive)?

Something like:
`// TypeScript and ESLint disagree about whether this can be null or undefined due to the way the this package links to nimble-components`